hack/generator/pkg/astmodel/strip_unused.go

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import (
+	"k8s.io/klog/v2"
+)
+
+// TypeNameSet stores type names in no particular order without
+// duplicates.
+type TypeNameSet map[TypeName]bool
+
+// NewTypeNameSet makes a TypeNameSet containing the specified
+// names. If no elements are passed it might be nil.
+func NewTypeNameSet(initial ...TypeName) TypeNameSet {
+	var result TypeNameSet
+	for _, name := range initial {
+		result = result.Add(name)
+	}
+	return result
+}
+
+// Add includes the passed name in the set and returns the updated
+// set, so that adding can work for a nil set - this makes it more
+// convenient to add to sets kept in a map (in the way you might with
+// a map of slices).
+// TODO(babbageclunk): I like the idea of making the zero element
```

Comment:
I'm not sure about this - it makes the map update code pretty nice, but might be too confusing.